File: backend/app/models/clip.py
from datetime import datetime

from sqlalchemy import (
    Column,
    DateTime,
    Float,
    ForeignKey,
    Integer,
    String,
)
from sqlalchemy.orm import relationship

from app.db.database import Base


class Clip(Base):
    __tablename__ = "clips"

    id = Column(Integer, primary_key=True, index=True)

    video_id = Column(
        Integer,
        ForeignKey("videos.id"),
        nullable=False,
    )

    title = Column(
        String,
        nullable=False,
    )

    start_time = Column(
        Float,
        nullable=False,
    )

    end_time = Column(
        Float,
        nullable=False,
    )

    clip_path = Column(
        String,
        nullable=True,
    )

    subtitle_path = Column(
        String,
        nullable=True,
    )

    thumbnail_path = Column(
        String,
        nullable=True,
    )

    status = Column(
        String,
        nullable=False,
        default="PENDING",
    )

    retry_count = Column(
        Integer,
        nullable=False,
        default=0,
    )

    error_message = Column(
        String,
        nullable=True,
    )

    editing_style = Column(
        String,
        nullable=False,
        default="AUTO",
    )

    applied_preset = Column(
        String,
        nullable=True,
    )

    ai_style_recommendation = Column(
        String,
        nullable=True,
    )

    ai_style_confidence = Column(
        Float,
        nullable=True,
    )

    ai_style_reason = Column(
        String,
        nullable=True,
    )

    created_at = Column(
        DateTime,
        default=datetime.utcnow,
    )

    video = relationship(
        "Video",
        back_populates="clips",
    )

    publications = relationship(
        "Publication",
        back_populates="clip",
        cascade="all, delete-orphan",
    )

File: frontend/src/types/clip.ts
export interface Clip {
  id: number;
  video_id: number;
  title: string;
  start_time: number;
  end_time: number;
  duration: number;
  status: string;
  thumbnail_url: string;
  stream_url: string;
  download_url: string;
  publication_url?: string | null;
  publication_id?: number | null;
  publication_platform?: string | null;
  publication_status?: string | null;
  publication_scheduled_at?: string | null;
  publication_error_type?: string | null;
  publication_error_details?: string | null;
  editing_style?: string;
  applied_preset?: string | null;
  ai_style_recommendation?: string | null;
  ai_style_confidence?: number | null;
  ai_style_reason?: string | null;
}
